src/providers/mbed_provider/key_management.rs

```diff
@@ -212,7 +217,9 @@ impl MbedProvider {
         let export_length = psa_crypto_key_management::export_public(id, &mut buffer)?;
 
         buffer.resize(export_length, 0);
-        Ok(psa_export_public_key::Result { data: buffer })
+        Ok(psa_export_public_key::Result {
+            data: buffer.into(),
```

Comment:
Would these changes need to be made in the other providers too?

Reply:
Probably. I can't test all providers, resolving merge conflict and making tweaks for your first comment so Github can run CI.

Reply:
Setup my VM for TPM now so it'll build.

You're right, TPM and PKCS11 providers need updating. Hit a problem with TPM - it uses `SignatureData::EcdsaSignature` from `tss-esapi` which uses `Vec<u8>` to store the two parts of the signature. Do we have to accept that those can't be changed to `Zeroize`, or do we have other options?

Reply:
Ah, you and @hug-dev are the authors of that, so I guess we don't have to just accept it 🎉

Reply:
> Do we have to accept that those can't be changed to Zeroize, or do we have other options?

That can be done in a best-effort way for now, zeroizing when/where we can, and we can go and update the `tss-esapi` crate to help with that later. Ideally we should be treating this data consistently across the stack (well, the parts that we control anyway). After I'm done with the client I'll look at zeroizing in the TPM crate.